src/classes/ADDR_CannotDelete_TDTM.cls

```diff
@@ -0,0 +1,90 @@
+    modification, are permitted provided that the following conditions are met:
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Addresses
+* @group-content ../../ApexDocContent/Addresses.htm
+* @description Stops an Address from being deleted if it has any Account or Contact child records. 
+*/
+public with sharing class ADDR_CannotDelete_TDTM extends TDTM_Runnable {
+    /*******************************************************************************************************
+    * @description Get the setting of preventing Address deletion
+    */
+    private static Boolean enabledPreventAddressDeletion = UTIL_CustomSettingsFacade.getSettings().Prevent_Address_Deletion__c;
+    
+    /*******************************************************************************************************
+    * @description Stops an Address from being deleted if it has any Account or Contact child records. 
+    * @param newlist the list of Addresses from trigger new. 
+    * @param oldlist the list of Addresses from trigger old. 
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.). 
+    * @param objResult the describe for Addresses 
+    * @return DmlWrapper.  
+    ********************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
+    TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+        
+        if (!enabledPreventAddressDeletion) {
```

Comment:
```suggestion
        if (!ADDR_CannotDelete_TDTM.enabledPreventAddressDeletion) {
```
Please add the classname before referencing a static variable so we may have that reference explicitly understood.

Reply:
@MatthewBlanski would it make the code more readable to replace the ! with a check for FALSE instead?  
if (ADDR_CannotDelete_TDTM.enabledPreventAddressDeletion == FALSE) {

Reply:
@m-kinkade I believe it would. When the check gets that long, it can be helpful. I wasn't going to list that as a required change.